src/nvim/msgpack_rpc/helpers.c

```diff
@@ -380,12 +380,11 @@ void msgpack_rpc_from_object(const Object result, msgpack_packer *const res)
   kv_push(stack, ((APIToMPObjectStackItem) { &result, false, 0 }));
   while (kv_size(stack)) {
     APIToMPObjectStackItem cur = kv_last(stack);
-    STATIC_ASSERT(kObjectTypeWindow == kObjectTypeBuffer + 1
-                  && kObjectTypeTabpage == kObjectTypeWindow + 1,
+    STATIC_ASSERT(kObjectTypeTabpage == kObjectTypeWindow + 1,
                   "Buffer, window and tabpage enum items are in order");
```

Comment:
Why did the IDE suggest this? It should be explained in the commit message, it is not obvious.

Reply:
@justinmk ack. I will see if I can find a good explanation for this.

Reply:
So the explanation is simple. If you look at `defs.h`:

```
typedef enum {
  kObjectTypeNil = 0,
  kObjectTypeBoolean,
  kObjectTypeInteger,
  kObjectTypeFloat,
  kObjectTypeString,
  kObjectTypeArray,
  kObjectTypeDictionary,
  kObjectTypeLuaRef,
  // EXT types, cannot be split or reordered, see #EXT_OBJECT_TYPE_SHIFT
  kObjectTypeBuffer,
  kObjectTypeWindow,
  kObjectTypeTabpage,
} ObjectType;
```

`kObjectTypeBuffer` is immediately followed by `kObjectTypeWindow` and since C enums are ints
`kObjectTypeWindow == kObjectTypeBuffer + 1` is always true

Reply:
@justinmk do you want to include this change?

Reply:
Well,  `STATIC_ASSERT`  guards against potential future changes to distant parts of the source code. They are often redundant, intentionally. I think it shouldn't be changed.